godot-core/src/obj/raw_gd.rs

```diff
@@ -522,10 +522,10 @@ where
         if T::DynMemory::pass_as_ref(call_type) {
             interface_fn!(ref_set_object)(ptr as sys::GDExtensionRefPtr, self.obj_sys())
         } else {
-            ptr::write(ptr as *mut _, self.obj)
+            ptr::write(ptr as *mut _, self.obj);
+            // We've passed ownership to caller.
+            std::mem::forget(self);
         }
-        // We've passed ownership to caller.
-        std::mem::forget(self);
```

Comment:
@Bromeon this fixes the leak. Calling godots `ref_set_object` creates a new ref counted reference in the engine and increments the count by 1. Since we already hold the initial reference, the count is now at 2. Calling `mem::forget(self)` on our reference prevents the count from going down to 1 where it should be after we return from our function.

Reply:
Thanks a lot, great find!

If it's not too much to ask, could you extract this commit under its own PR? Reason is that we can then later link it as bugfix in the changelog.

Also, maybe add as a comment what you just described here for `ref_set_object`?

Reply:
yes, I will extract it.